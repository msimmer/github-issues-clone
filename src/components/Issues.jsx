// @flow

import type { Repository, IssueNode } from "../types";

import * as React from "react";
import Issue from "./Issue";

const Issues = (props: { repository: Repository }): React.Node => {
  const { edges } = props.repository.issues;
  if (edges.length < 1) return null;
  return edges.map(({ node }: IssueNode) => <Issue key={node.id} {...node} />);
};

export default Issues;
