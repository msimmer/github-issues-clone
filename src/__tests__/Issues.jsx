import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Issues from "../components/Issues";

configure({ adapter: new Adapter() });

const mockProps = {
  repository: {
    issues: {
      edges: [
        {
          node: {
            id: 1
          }
        }
      ]
    }
  }
};

describe("<Issues />", () => {
  it("matches the snapshots", () => {
    const component = shallow(<Issues {...mockProps} />);
    expect(toJSON(component)).toMatchSnapshot();
  });
});
