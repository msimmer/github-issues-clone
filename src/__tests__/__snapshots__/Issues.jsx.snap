// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Issues /> matches the snapshots 1`] = `
<Issue
  id={1}
  key="1"
/>
`;
